swan-lake/learn/security/authentication-and-authorization.md

```diff
@@ -1,530 +1,309 @@
     secureSocket: {
         keyStore: {
-            path: config:getAsString("b7a.home") + "/bre/security/ballerinaKeystore.p12",
+            path: "/path/to/ballerinaKeystore.p12",
             password: "ballerina"
         }
     }
 });
 
 @http:ServiceConfig {
-    basePath: "/hello"
+    auth: [
+        {
+            jwtValidatorConfig: {
+                issuer: "wso2",
+                audience: "ballerina",
+                trustStoreConfig: {
+                    trustStore: {
+                        path: "/path/to/ballerinaTruststore.p12",
+                        password: "ballerina"
+                    },
+                    certificateAlias: "ballerina"
+                },
+                scopeKey: "scp"
+            },
+            scopes: ["hello"]
+        }
+    ]
 }
-service helloWorld on secureHelloWorldEp {
-
-    @http:ResourceConfig {
-        methods: ["GET"],
-        path: "/"
-    }
-    resource function sayHello(http:Caller caller, http:Request req) {
-        http:Response resp = new;
-        resp.setTextPayload("Hello, World!");
-        checkpanic caller->respond(resp);
+service /foo on securedEP {
+    resource function get bar() returns string {
+        return "Hello, World!";
     }
 }
 ```
 
 When the service is invoked without authentication information or invalid authentication information, an authentication failure will occur:
 
 ```
-curl -k -v https://localhost:9091/hello
+curl -k -v https://localhost:9090/foo/bar
 
-> GET /hello HTTP/1.1
-> Host: localhost:9091
+> GET /foo/bar HTTP/1.1
+> Host: localhost:9090
 > User-Agent: curl/7.47.0
 > Accept: */*
 > 
 < HTTP/1.1 401 Unauthorized
+< content-length: 0
+< server: ballerina
 < content-type: text/plain
 < 
-Authentication failure
 ```
 
-Although a request is made with valid authentication information, if the introspection endpoint does not respond either with the `scope` attribute of the response JSON payload or with an unexpected `scope` attribute, an authorization failure will occur.
+If a request is made with valid authentication information, but the authenticated user does not have the required permission, an authorization failure will occur.
 
 ```
-curl -k -v https://localhost:9091/hello -H "Authorization:Bearer <token>"
+curl -k -v https://localhost:9090/foo/bar -H "Authorization: Bearer <token>"
 
-> GET /hello HTTP/1.1
-> Host: localhost:9091
+> GET /foo/bar HTTP/1.1
+> Host: localhost:9090
 > User-Agent: curl/7.47.0
 > Accept: */*
-> Authorization:Bearer <token>
+> Authorization: Bearer <token>
 >
 
 < HTTP/1.1 403 Forbidden
+< content-length: 0
+< server: ballerina
 < content-type: text/plain
 <
-Authorization failure
 ```
 
-A request, which gets a successful response from the introspection endpoint with a correct "scope" attribute will result in a successful invocation.
+A request is made with valid authentication information, and the authenticated user has the required permission, will result in a successful invocation.
```

Comment:
```suggestion
When a request is made with valid authentication information, if the authenticated user has the required permission, this will result in a successful invocation.
```